tools/python_tools/cuvslam_tools/tests/test_icl_nuim_conversion.py

```
@@ -0,0 +1,401 @@
+        second = [float(value) for value in rows[1].split()]
+        self.assertAlmostEqual(second[3], 0.1, places=9)
+
+    def test_posed_frame_with_no_association_is_reported(self):
+        source = self._write_source(image_indices=(1, 2), pose_indices=(2, 3))
+        with self.assertRaisesRegex(convert_icl_nuim.ConversionError, "no entry for 1 posed"):
+            convert_icl_nuim.convert_sequence(source, self.spec, self.root / "out")
+
+    def test_missing_media_file_is_reported(self):
+        source = self._write_source()
+        (source / "rgb" / "3.png").unlink()
+        with self.assertRaisesRegex(convert_icl_nuim.ConversionError, "missing 1 media file"):
```

Ruff (0.16.3)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)